tectonic-stats-extender/test/integration_test.go

```diff
@@ -0,0 +1,300 @@
+package test
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/api/iterator"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	// These consts are extension name's we want to ensure are present in
+	// BigQuery results.
+	accountIDExtension              = "accountID"
+	certificatesStrategyExtension   = "certificatesStrategy"
+	installerPlatformExtension      = "installerPlatform"
+	tectonicUpdaterEnabledExtension = "tectonicUpdaterEnabled"
+	// extensionsNameKey is the key for extension names in BigQuery results.
+	extensionsNameKey = "extensions_name"
+	// extensionsValueKey is the key for extension values in BigQuery results.
+	extensionsValueKey = "extensions_value"
+	// tectonicSystemNamespace is the namespace for Tectonic.
+	tectonicSystemNamespace = "tectonic-system"
+)
+
+var (
+	// timeout is the maximum time for a test.
+	timeout time.Duration
+	// bigQuerySpec is the spec of the BigQuery table to test for cluster metrics.
+	bigQuerySpec string
+)
+
+func TestMain(m *testing.M) {
+	flag.DurationVar(&timeout, "timeout", 1*time.Minute, "maximum time for a test (default 1m)")
+	flag.StringVar(&bigQuerySpec, "bigqueryspec", "", "BigQuery Spec (formatted as `bigquery://project.dataset.table`)")
+	flag.Parse()
+
+	os.Exit(m.Run())
+}
+
+// Test is the only test suite run by default.
+func Test(t *testing.T) {
+	t.Run("StatsEmitterLogs", testGetStatsEmitterLogs)
+	t.Run("BigQueryData", testGetBigQueryData)
+}
+
+// newClient will attempt to get produce a client from a config file
+// specified in the KUBECONFIG environment variable. If this environment
+// variable is empty, then `BuildConfigFromFlags` automatically tries to
+// get a config from the in-cluster configuration.
```

Comment:
This is perfect for the future, in case we ever want to run the container as a pod inside the cluster.